backend/server.js

```diff
@@ -1,29 +1,147 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+import crypto from "crypto";
+import bcrypt from "bcrypt";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI";
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true, // To get rid of deprecation warning regarding collection.ensureIndex
+  useFindAndModify: false // To get rid of deprecation warning regarding findOneAndUpdate()
+});
+mongoose.Promise = Promise;
+
+const userSchema = new mongoose.Schema({
+  alias: {
+    type: String,
+    default: ""
+  },
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+    minlength: 5
+  },
+  password: {
+    type: String,
+    required: true,
+    minlength: 5
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString("hex"),
+    unique: true
+  }
+});
+
+userSchema.pre("save", async function (next) {
+  const user = this;
+  if (!user.isModified("password")) {
+    return next();
+  }
+  const salt = bcrypt.genSaltSync();
+  user.password = bcrypt.hashSync(user.password, salt);
+  next();
+});
+
+const User = mongoose.model("User", userSchema);
+
+const authenticateUser = async (req, res, next) => {
+  try {
+    const user = await User.findOne({ accessToken: req.header('Authorization') });
+    req.user = user;
+    next();
+  } catch (err) {
+    res.status(401).json({ loggedOut: true });
+  }
+};
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+const listEndpoints = require("express-list-endpoints");
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+const allowedDomains = ["https://max-sandrine-auth-app.netlify.app", "http://localhost:3000"];
+app.use(cors({
+  origin: function (origin, callback) {
+    if (!origin) return callback(null, true);
+    if (allowedDomains.indexOf(origin) === -1) {
+      var msg = `This site ${origin} does not have an access. Only specific domains are allowed to access it`
```

Comment:
Nice that you implement this logic, but remember that you shouldn't use `var` any more. `const` would be proper here. 🤓